File: 08-1-2-State-Solution/Partner.cs
namespace State
{
    using States;

    public class Partner
    {
        internal IPartnerState State { get; set; }

        public Partner()
        {
            State = new NeutralState(this);
        }

        public void Talk() => State.Talk();

        public void Kiss() => State.Kiss();

        public void Insult() => State.Insult();
    }
}

File: 09-DesignPatterns-Java/Solution/State/src/Partner/Partner.java
package Partner;

import States.IPartnerState;
import States.NeutralState;

public class Partner {
    private IPartnerState _state = new NeutralState(this);

    public void setState(IPartnerState state) {
        _state = state;
    }

    public void talk() {
        _state.talk();
    }

    public void kiss() {
        _state.kiss();
    }

    public void insult() {
        _state.insult();
    }
}
